OData/test/E2ETest/WebStack.QA.Test.OData/ODataPathHandler/LinkGenerationTests.cs

```diff
@@ -89,5 +105,19 @@ public void GeneratedLinkShouldMatchRequestRouting()
             content = this.Client.GetStringAsync(this.BaseAddress + "/v2/LinkGeneration_Model2").Result;
             Assert.DoesNotContain(@"/v1/LinkGeneration_Model2", content);
         }
+
+        [Theory]
+        [InlineData("/v1/LinkGeneration_Model1(1)/NonContainedNavigationProperty", "/v1/LinkGeneration_Model2(2)")]
```

Comment:
Only one test case, it's better to use [Fact] and put the cast condition into test body.